joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -208,17 +209,18 @@ JointTrajectoryController::update()
             RCLCPP_WARN(node_->get_logger(), "Aborted due to goal tolerance violation");
             result->set__error_code(FollowJTrajAction::Result::GOAL_TOLERANCE_VIOLATED);
           }
-          rt_active_goal_->setAborted(result);
-          rt_active_goal_.reset();
-        }
+          active_goal->setAborted(result);
+          // TODO(matthew-reynolds): Need a lock-free write here
```

Comment:
Please create a follow-up issue with a short write-up and insert the link of that here (and other instances) instead. Much easier to follow up on it that way.

Reply:
Done, see #168 and #169